lib/index.js

```diff
@@ -290,3 +295,70 @@ internals.stop = async (server) => {
 
 internals.state = (realm) => Toys.state(realm, 'schwifty');
 internals.rootState = (realm) => Toys.rootState(realm, 'schwifty');
+
+internals.migrateDiffCmd = async (srv, args, rootDir, ctx) => {
+
+    const SchwiftyMigrateDiff = SchwiftyMigrateDiffHelper.get();
+
+    if (!SchwiftyMigrateDiff) {
+        throw new ctx.DisplayError('To use this command please first install `npm install --save-dev schwifty-migrate-diff`');
+    }
+
+    const migrationName = args.shift();
+    if (!migrationName || migrationName.startsWith('--')) {
+        throw new ctx.DisplayError('A name must be specified for the generated migration');
+    }
+
+    const definition = {
+        mode: {
+            description: 'Set mode to "create" or "alter" (defaults to "alter")'
+        },
+        dir: {
+            description: 'Set migrationsDir'
+        }
+    };
+
+    const parsedArgs = ctx.Bossy.parse(definition, { argv: args });
```

Comment:
Passing bossy through means that the command would depend on a certain version of hpal, one big downside of leaning on `ctx` in general.  (This is partly why I only documented `DisplayError` as being officially public.)